--- ./core/httpd-nonos.c
+++ ./core/httpd-nonos.c
@@ -9,10 +9,15 @@ ESP8266 web server - platform-dependent routines, nonos version
 
 #ifndef FREERTOS
 
+#define PACKET_COUNT_RX 1
+#define PACKET_COUNT_TX 2
+
 //Listening connection data
 static struct espconn httpdConn;
 static esp_tcp httpdTcp;
 
+extern void ICACHE_FLASH_ATTR packet_counter(ConnTypePtr con, uint8_t direction);
+
 //Set/clear global httpd lock.
 //Not needed on nonoos.
 void ICACHE_FLASH_ATTR httpdPlatLock() {
@@ -37,11 +42,13 @@ static void ICACHE_FLASH_ATTR platDisconCb(void *arg) {
 
 static void ICACHE_FLASH_ATTR platRecvCb(void *arg, char *data, unsigned short len) {
 	ConnTypePtr conn=arg;
+	packet_counter(conn, PACKET_COUNT_RX);
 	httpdRecvCb(conn, (char*)conn->proto.tcp->remote_ip, conn->proto.tcp->remote_port, data, len);
 }
 
 static void ICACHE_FLASH_ATTR platSentCb(void *arg) {
 	ConnTypePtr conn=arg;
+	packet_counter(conn, PACKET_COUNT_TX);
 	httpdSentCb(conn, (char*)conn->proto.tcp->remote_ip, conn->proto.tcp->remote_port);
 }
